src/Oscoin/Crypto/Address/Bech32.hs

```diff
@@ -0,0 +1,58 @@
+{-- | Partial, opinionated support for Bech32.
+-- Mostly taken and adapted from the reference implementation available
+-- <here https://github.com/sipa/bech32/blob/master/ref/haskell/src/Codec/Binary/Bech32.hs>
+--}
+module Oscoin.Crypto.Address.Bech32
+    ( createChecksum
+    , verifyChecksum
+
+    -- * Interals
+    , createBech32checksum
+    , verifyBech32checksum
+    ) where
+
+import           Oscoin.Prelude
+
+import           Data.Bits
+import qualified Data.ByteString as BS
+import           Oscoin.Crypto.Address.Internal (Checksum(..))
+
+(.>>.), (.<<.) :: Bits a => a -> Int -> a
+(.>>.) = unsafeShiftR
+(.<<.) = unsafeShiftL
+
+type HumanReadablePart = ByteString
```

Comment:
Note sure what this type synonym gives us - It makes the function signatures slightly more confusing to me.

Reply:
we can surely scrap it -- what it gives you is essentially a bit of consistency with the original Bench32 spec, as they explicitly introduces the concept of "HumanReadablePart". I'm not attached to it, so I'm gonna scrap it 😬

Reply:
84c5a8f